src/logic/calculate.js

```diff
@@ -1,20 +1,52 @@
 import operate from './operate';
 
-function Calculate(calculatorData, buttonName) {
-  // eslint-disable-next-line prefer-const
+function Calculate(buttonName, calculatorData) {
   let { total, next, operation } = calculatorData;
-  const dataNumbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
-  const dataOperations = ['+', '-', 'x', '÷', '%'];
+  const dataNumbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '.'];
+  const dataOperations = ['+', '-', 'x', '%', '÷'];
+
+  const count = (elem, arr) => {
+    let sum = 0;
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < arr.length; i++) {
+      if (arr[i] === elem) {
+        sum += 1;
+      }
+    }
+    return sum;
+  };
 
   if (buttonName === '+/-') {
-    total *= -1;
+    total = parseFloat(total) * (-1).toString();
   } else if (dataNumbers.includes(buttonName)) {
-    total += buttonName;
+    if (total === null || total === 'Infinity') {
+      total = buttonName;
+    } else if (buttonName === '.') {
```

Comment:
I notice whenever we click `.` we have just `.` instead of having `0.`
example: we click `.` => `.5x1` which supposed to be `0.5x1`
I trust you can make the little adjustment before merging 👍